### PR TITLE
ENH: PR: Alternate extra wall

### PR DESCRIPTION
**Pull request by igiannakas**
**Ported Cura feature - Alternate Extra wall.**
https://support.makerbot.com/s/article/1837650983225

This setting adds an extra wall every other layer. This way the infill gets caught between the walls, resulting in:
1. Theoretically, stronger prints due to better infill anchoring.
2. Allows for the infill/wall overlap to be set to low values without compromising object strength while maintaining external surface quality.
3. It may contribute to improving water tightness of models as the gaps between the top and bottom layers and the walls are overlapping.

This option overrides the ensure vertical thickness algorithm as this would simply create infill lines in-between the layers instead of allowing the infill to anchor in-between the layers.

image

### CODE REVIEW

**Comment by Eldenroot** [comment]
Thank you!

**Comment by igiannakas** [comment]
> Just minor typo in the tooltip. Not a native speaker

Fixed ;)

**Comment by SoftFever** [comment]
@igiannakas 
Fantastic work!
I have a suggestion: since it overrides/invalids ensure_vertical_shell_thickness, it would be better to pop up a dialog to warn the user and disable ensure_vertical_shell_thickness when activating `Alternate extra wall`.
Similar to the vase mode option.

**Comment by igiannakas** [comment]
> @igiannakas Fantastic work! I have a suggestion: since it overrides/invalids ensure_vertical_shell_thickness, it would be better to pop up a dialog to warn the user and disable ensure_vertical_shell_thickness when activating `Alternate extra wall`. Similar to the vase mode option.

I was debating doing so - I’ll update the code

**Comment by igiannakas** [comment]
Done - dialog box introduced.

File: src/slic3r/GUI/ConfigManipulation.cpp [hunk 1]
```diff
@@ -282,7 +283,7 @@ void ConfigManipulation::update_print_fff_config(DynamicPrintConfig* config, con
            ! config->opt_bool("overhang_reverse") &&
             config->opt_enum<TimelapseType>("timelapse_type") == TimelapseType::tlTraditional))
     {
-        wxString msg_text = _(L("Spiral mode only works when wall loops is 1, support is disabled, top shell layers is 0, sparse infill density is 0 and timelapse type is traditional."));
+        wxString msg_text = _(L("Spiral mode only works when wall loops is 1, alternate extra wall is disabled, support is disabled, top shell layers is 0, sparse infill density is 0 and timelapse type is traditional."));
```
Comment (SoftFever) [inline src/slic3r/GUI/ConfigManipulation.cpp:286]: I'm thinking about this.
I think we should do otherway around, when spiral vase is used we should skip extra wall in the code(PerimeterGenerator class) automatically.
PerimeterGenerator has been already checking spiral mode(by using m_spiral_vase), so we should just avoid the loop+1 logic when m_spiral_vase is true.

Let me know what you think.
```suggestion
        wxString msg_text = _(L("Spiral mode only works when wall loops is 1, alternate extra wall is disabled, support is disabled, top shell layers is 0, sparse infill density is 0 and timelapse type is traditional."));
```

Comment (igiannakas) [reply src/slic3r/GUI/ConfigManipulation.cpp:286]: no preference from my side - either works ;) ill update the pr soon

Comment (igiannakas) [reply src/slic3r/GUI/ConfigManipulation.cpp]: Done. Also I removed the ensure vertical shell thickness check in vase mode - this is not needed for sure and it was clashing with the alternate extra wall pop up that I just implemented in the change above ;)

**Review by SoftFever — APPROVED** [review]
Looks good

**Comment by SoftFever** [comment]
Merged.
Thank you very much 👍

**Comment by jlo1975** [comment]
hello
By default when ticking the "alternate extra wall" the "ensure vertical shell thickness" is set to none (as stated in the warning popup).
This result in some print to not have missing bottom layers.
But you can select "ensure vertical shell thickness"  to moderate or critical part only and you have the best of both worlds.

I suggest to change the default selection of "ensure vertical shell thickness" from none to moderate when selecting "alternate extra wall"

**Comment by jlo1975** [comment]
image

image

**Comment by jlo1975** [comment]
on screenshot above you can see we can still be conservative on vertical shell thickness even with alternate extra wall activated

**Comment by jlo1975** [comment]
and yes we still have the extra wall 

image

**Comment by admiralakber** [comment]
> I suggest to change the default selection of "ensure vertical shell thickness" from none to moderate when selecting "alternate extra wall"

I support @jlo1975's suggestion here. Before I read their post and had "ensure vertical shell thickness" to None - top layers on multiple of my prints were severely under extruded - it wasn't even a single top layer. The bottom layer was also thinner and had missing bottom shell layers.

I am using OrcaSlicer 2.0 Beta on Linux.

**Comment by jlo1975** [comment]
> > I suggest to change the default selection of "ensure vertical shell thickness" from none to moderate when selecting "alternate extra wall"
> 
> I support @jlo1975's suggestion here, or at least set it to critical by default. Before I read their post and had "ensure vertical shell thickness" to None - top layers on multiple of my prints were severely under extruded - it wasn't even a single top layer. The bottom layer was also thinner and had missing bottom shell layers.
> 
> I am using OrcaSlicer 2.0 Beta on Linux.

for info the latest build has a bugfix for the "none" setting where the bottom/top surface are now having their layers.
So we may consider my suggestion was a workaround but I think letting to "moderate" when activating the extra wall feature is a good trade offf

**Comment by SoftFever** [comment]
Yeah, it's bug.
There are two issue in the beta
1. None shouldn't generate only one top/bottom layer
2. Turn on alternate extra wall shouldn't change vertical thickness option to None.(this regression was introduced in another PR)

Both issues have been fixed in nightly build.

**Pull request merged**